content/docs/practices/naming.md

```diff
@@ -20,30 +20,40 @@ A metric name...
-* ...should have a suffix describing the unit, in plural form. Note that an accumulating count has `total` as a suffix, in addition to the unit if applicable.
- * <code>http\_request\_duration\_<b>seconds</b></code>
- * <code>node\_memory\_usage\_<b>bytes</b></code>
- * <code>http\_requests\_<b>total</b></code>
-   (for a unit-less accumulating count)
- * <code>process\_cpu\_<b>seconds\_total</b></code>
-   (for an accumulating count with unit)
- * <code>foobar_build<b>\_info</b></code>
-   (for a pseudo-metric that provides [metadata](https://www.robustperception.io/exposing-the-software-version-to-prometheus) about the running binary)
- * <code>data\_pipeline\_last\_record\_processed\_<b>timestamp_seconds</b></code>
-  (for a timestamp that tracks the time of the latest record processed in a data processing pipeline)
+* ...should use base units (e.g. seconds, bytes, meters - not milliseconds, megabytes, kilometers).See below for a list of base units.
+* ...should have a suffix describing the unit, in plural form. Note that an accumulating count has `total` as a suffix, in addition to the unit if applicable. Also note that this applies to units in the narrow sense (like the units in the table below), but not to countable things in general. For example, <code>connections</code> or <code>notifications</code> are not considered units for this rule and do not have to be at the end of the metric name. (See also examples in the next paragraph.)
+  * <code>http\_request\_duration\_<b>seconds</b></code>
+  * <code>node\_memory\_usage\_<b>bytes</b></code>
+  * <code>http\_requests\_<b>total</b></code>
+    (for a unit-less accumulating count)
+  * <code>process\_cpu\_<b>seconds\_total</b></code>
+    (for an accumulating count with unit)
+  * <code>foobar_build<b>\_info</b></code>
+    (for a pseudo-metric that provides [metadata](https://www.robustperception.io/exposing-the-software-version-to-prometheus) about the running binary)
+  * <code>data\_pipeline\_last\_record\_processed\_<b>timestamp_seconds</b></code>
+    (for a timestamp that tracks the time of the latest record processed in a data processing pipeline)
+* ...may order its name components in a way that leads to convenient grouping when a list of metric names is sorted lexicographically, as long as all the other rules are followed. The following examples have their the common name components first so that all the related metrics are sorted together:
+  * <code>prometheus\_tsdb\_head\_truncations\_closed\_total</code>
+  * <code>prometheus\_tsdb\_head\_truncations\_established\_total</code>
+  * <code>prometheus\_tsdb\_head\_truncations\_failed\_total</code>
+  * <code>prometheus\_tsdb\_head\_truncations\_total</code>
+  The following examples are also valid, but are following a different trade-off. They are easier to read individually, but unrelated metrics like <code>prometheus\_tsdb\_head\_series</code> might get sorted in between.
```

Comment:
```suggestion

    The following examples are also valid, but are following a different trade-off. They are easier to read individually, but unrelated metrics like <code>prometheus\_tsdb\_head\_series</code> might get sorted in between.
```

I think we need a blank line here to convince markdown to end the previous list item. (I'll commit this directly, no need to change the PR.)